Add new request page

sharrief opened the pull request:
Adding react scaffolding for the known required fields and form per feature 1.2.2.1
Tests incoming

<img width="1467" alt="image" src="https://github.com/digitalaidseattle/seattle-humane-app/assets/17752660/dfdd0e2d-8999-4a85-b5c3-3bc3234c9197">

jnakaso commented:
You should update/replace /pages/client/index.tsx

sharrief commented:
> You should update/replace /pages/client/index.tsx.
> 
> Consider linking new client requests from the Clients table /pages/clients/index.tsx

Thanks. I was using that as a placeholder. I think this is supposed to be a root nav modal so a worker can open the modal from any page. I'm working on adding it to the root nav menu, but we don't want more than one modal open at a time.

I am considering how to integrate it with the existing entities and service calls, but I need to check with Sam if that's in the scope he was planning on taking on when he suggested splitting this feature into two work items, one frontend and one backend.

jnakaso commented on src/components/FormConfirmationButtons.tsx:
let's use optional parameters for cancel & save. This will promote consistent labeling and reduce burden on other users of this component.

sharrief replied:
Good idea, thanks!

jnakaso commented:
> > You should update/replace /pages/client/index.tsx.
> > Consider linking new client requests from the Clients table /pages/clients/index.tsx
> 
> Thanks. I was using that as a placeholder. I think this is supposed to be a root nav modal so a worker can open the modal from any page. I'm working on adding it to the root nav menu, but we don't want more than one modal open at a time.
> 
> I am considering how to integrate it with the existing entities and service calls, but I need to check with Sam if that's in the scope he was planning on taking on when he suggested splitting this feature into two work items, one frontend and one backend.

There is also /src/components/ClientDialog.tsx  that is invoked from the app tool bar that would need to be updated.  That should be updated.

sharrief commented:
Thanks for the comments. I've updated the ClientDialog to use the same new form/fields. I also added configurable visibility for form fields so the ClientDialog shows the same fields. 

We'll need some work on the data service side to enable pet/staff lookup controls. That scenario wasn't handled before, and I think it's out of scope for this work item.

jnakaso commented on src/types.ts:
```diff
@@ -11,22 +11,34 @@ export type ClientType = {
\ No newline at end of file
+}
+// TODO use lookup with assignedTo
+export type EditableRequestType = Omit<RequestType, 'id'|'animal_id'|'staff_id'> & { assingedTo: string }
```
typo "assign"

jnakaso commented on src/context/serviceRequest/serviceInformationContext.tsx:
```diff
@@ -0,0 +1,33 @@
+  source: '',
+  description: '',
+  status: '',
+  assingedTo: ''
```
typo

jnakaso commented on src/types.ts:
```diff
@@ -11,22 +11,34 @@ export type ClientType = {
   last_name: string;
   email: string;
   phone: string;
+  postalCode: string;
```
camelCase vs underscore?

sharrief replied:
Ah yes, assignedTo is also camelCase. Will normalize on underscore

jnakaso reviewed (approved):
Looks good.
